app/controllers/login_controller.rb

```diff
@@ -7,6 +7,8 @@ class LoginFailedError < StandardError; end
 class LoginController < ApplicationController
   before_action :authenticate_user
   before_action :check_for_read_only_mode, :except => [:index]
+  before_action :require_no_user_or_redirect,
+                only: [:index, :login, :forgot_password, :reset_password]
```

Comment:
Should `:only => [:index, :login, :forgot_password, :reset_password]` be used here rather than `only: [:index, :login, :forgot_password, :reset_password]`?

I checked the rails documentation on archive.org and it seems that `:only =>` is the older version of `only:`

cc @pushcx

Reply:
Ruby hashes with symbol keys used to look like this `{ :only => [] }`. A few years after the Lobsters codebase started, Ruby added a new style for this, `{ only: [] }` in 2007 and it took a long time to become popular. Some people don't like the inconsistency that Ruby requires `=>` for any key that's not a symbol.

The Lobsters codebase [does not consistently](https://github.com/lobsters/lobsters/pull/460/commits/69df28173823908708480da0dc7b6d8741268765) enforce [one or the other](https://www.rubydoc.info/gems/rubocop/0.46.0/RuboCop/Cop/Style/HashSyntax). It's not worth the churn to change them all. Try to use whichever is in use in that place or your personal preference if you're changing the line anyways.